client/dashboard/task-list/tasks/payments/index.js

```diff
@@ -226,6 +219,13 @@ class Payments extends Component {
 			);
 		}
 
+		let recommendedMethod = 'stripe';
+		methods.forEach( ( method ) => {
```

Comment:
Can we move this check into `componentDidUpdate` so this doesn't occur on every render?

We can check if there is a difference in the visible methods between current and previous props.

Reply:
> We can check if there is a difference in the visible methods between current and previous props.

[getPaymentMethods](https://github.com/woocommerce/woocommerce-admin/blob/c7444a716f0d464779ec07b854d42c433777066e/client/dashboard/task-list/tasks/payments/methods.js#L33) returns a new array every time it's called, so the logic to check if the new list of visible methods is the same as the old visible methods would be as complex (if not more) than just recalculating `recommendedMethod` every time.

> Can we move this check into componentDidUpdate so this doesn't occur on every render?

Done in https://github.com/woocommerce/woocommerce-admin/pull/4345/commits/b482b15e80f4733dd1475252f5f8dece938c1862